docs/process.md

```diff
@@ -1,3 +1,31 @@
+Development Processes
+---------------------
+
+Landing PRs
+===========
+
+ * Even after the code of a PR is approved, it should only be landed if the
+   CI on github is green, or the failures are known intermittent things.
+ * If you see an approved PR of someone without commit access (that either
+   you or someone else approved), land it for them (after checking CI as
+   mentioned earlier).
+ * If you approve a PR by someone with commit access, if there is no urgency
+   then leave it for them to land. (They may have other PRs to land alongside
+   it, etc.)
+ * It is strongly recommended to land PRs with github's "squash" option, which
+   turns the PR into a single commit. This makes sense if the PR is small,
+   which is also strongly recommended. However, sometimes a merge may make
+   more sense, *if and only if*:
+    * The PR is not easily separable into a series of small PRs (e.g., review
+      must consider all the commits, either because the commits are hard to
+      understand by themselves, or because review of a later PR may influence
+      an earlier PR's discussion).
+    * The individual commits have value (e.g., they are easier to understand
+      one by one).
+    * The individual commits are compatible with bisection (i.e., all tests
+      should pass after each commit).
```

Comment:
Would be good to require PRs to not have merge commits, i.e. please rebase merge commits away from PRs?

Reply:
I strongly agree with this.

Even if you there can be case to be made for not squashing,  we still prefer rebase over merge to keep simple linear history.

@kripken I think you can explicitly disable (in the settings) the merge button and force either rebase to rebase+squash.   Commiters can of course override this from the command line but it would be great to prevent this option even showing up in the UI.

Reply:
+1

Reply:
@sbc100 thanks, yes, i found the option and disabled merge commits. Now the options should be only "squash" or "rebase and merge".